k8s-openapi-codegen/src/main.rs

```diff
@@ -99,7 +140,7 @@ fn main() -> Result<(), Error> {
 	result
 }
 
-fn run(supported_version: supported_version::SupportedVersion, out_dir_base: &std::path::Path, client: &reqwest::blocking::Client) -> Result<(), Error> {
+fn run(supported_version: supported_version::SupportedVersion, out_dir_base: &std::path::Path, client: &reqwest::blocking::Client, spec_url_override: Option<&str>) -> Result<(), Error> {
```

Comment:
Just take `spec_url: &str` and make the caller specify either the default from `SupportedVersion::spec_url` or the user one.